File: bac_portal_web/src/main/java/com/bac/portal/controller/ContentController.java
package com.bac.portal.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.bac.content.service.ContentService;
import com.bac.pojo.TbContent;
import com.bac.utils.BacResult;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/content")
public class ContentController {

    //注入远程服务对象
    @Reference(timeout = 10000000)
    private ContentService contentService;

    /*
     需求:根据内容分类id查询广告内容
      */
    @RequestMapping("/findContentListByCategoryId/{categroyId}")
    public List<TbContent> findContentListByCategoryId(@PathVariable Long categroyId) {
        //调用远程服务
        return contentService.findContentByCategoryId(categroyId);
    }
}

File: bac_search_web/src/main/webapp/js/controller/contentController.js
//控制层
app.controller('contentController', function ($scope, contentService) {



    /*
    需求:根据内容分类id查询广告内容
     */
       $scope.ContentList=[];
    //定义数组存储广告信息

    //读取列表数据绑定到表单中.
    $scope.findContentListByCategoryId = function (categroyId) {
        contentService.findContentListByCategoryId(categroyId).success(
            function (response) {
                //数组的角标存储对应的内容.
                $scope.ContentList[categroyId]=response;

            }
        );
    }


});
